src/app/client/src/app/modules/groups/components/activity/activity-search/activity-search.component.ts

```diff
@@ -239,9 +256,29 @@ export class ActivitySearchComponent implements OnInit, OnDestroy {
   }
 
   addActivity(event) {
-    const cdata = [{id: _.get(event, 'data.identifier'), type: 'Course'}];
+    const cdata = [{id: _.get(event, 'data.identifier'), type: _.get(event, 'data.contentType')}];
     this.addTelemetry('activity-course-card', cdata);
-    this.router.navigate(['/learn/course', _.get(event, 'data.identifier')], { queryParams: { groupId: _.get(this.groupData, 'id') } });
+
+    switch (_.get(event, 'data.contentType').toLowerCase()) {
```

Comment:
This should not be hardcoded like this.. There are only 3 types of details screen, 
1. For a collection of trackable:true => goto course details screen (course in this case)
2. For a collection of trackable=false => goto textbook details screen (Textbook)
3. For any non-collection content => goto resource details page (Resource, QuestionSet, etc..)

Reply:
how trackable is coming into this scenario ? we haven't had a discussion regarding trackable in this redirection ?